### PR TITLE
refactor: add empty v-bind warning(re #7973)

### PR DESCRIPTION
**Pull request by lyhper**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/compiler/parser/index.js [hunk 1]
```diff
@@ -528,6 +528,15 @@ function processAttrs (el) {
         name = name.replace(bindRE, '')
         value = parseFilters(value)
         isProp = false
+        if (
+          process.env.NODE_ENV !== 'production' &&
+          value.trim().length === 0
+        ) {
+          warn(
+            `${name}="${value}": ` +
+            'The v-bind expression should not be empty'
```
Comment (posva) [inline src/compiler/parser/index.js]: I think we can improve the message to something like _the value for a v-bind expression cannot be empty. Found in '${name}'_.
Let's wait for others feedback as I am not native

File: test/unit/modules/compiler/parser.spec.js [hunk 2]
```diff
@@ -508,6 +508,7 @@ describe('parser', () => {
     expect(ast.attrs[1].value).toBe('"field1"')
     expect(ast.props[0].name).toBe('value')
     expect(ast.props[0].value).toBe('msg')
+    expect('The v-bind expression should not be empty').toHaveBeenWarned()
```
Comment (posva) [inline test/unit/modules/compiler/parser.spec.js]: Let's extract this into one single test

File: src/compiler/parser/index.js (hunk 1 above)
Comment (lyhper) [reply src/compiler/parser/index.js]: thanks, I will do it later

**Comment by lyhper** [comment]
@posva Is it ok?

**Comment by lyhper** [comment]
thank you 😀

**Comment by sdras** [comment]
The warning message sounds good to me as a native speaker 👍

File: src/compiler/parser/index.js [hunk 3]
```diff
@@ -528,6 +528,14 @@ function processAttrs (el) {
         name = name.replace(bindRE, '')
         value = parseFilters(value)
         isProp = false
+        if (
+          process.env.NODE_ENV !== 'production' &&
```
Comment (wistcc) [inline src/compiler/parser/index.js]: I think we should check all environments instead of just production.

**Pull request merged**